agent/task_context.go

```diff
@@ -127,16 +144,18 @@ func (tc *taskContext) getTimeoutType() timeoutType {
 
 // makeTaskConfig fetches task configuration data required to run the task from the API server.
 func (a *Agent) makeTaskConfig(ctx context.Context, tc *taskContext) (*internal.TaskConfig, error) {
-	if tc.project == nil {
-		grip.Info("Fetching project config.")
-		err := a.fetchProjectConfig(ctx, tc)
-		if err != nil {
-			return nil, err
-		}
+	if tc.taskConfig != nil && tc.taskConfig.Project != nil && tc.taskConfig.Project.Identifier != "" {
```

Comment:
Since this function is itself making the `taskConfig`, is it sufficient to just check `tc.taskConfig != nil`? The tests presumably already have to populate `taskConfig` themselves.

Reply:
It was failing for a bunch of tests when the check was just tc.taskConfig != nil. If I remember correctly, some had the taskConfig defined but not the project and were relying on this fetching it.